govcd/api_token.go

```diff
@@ -21,122 +21,208 @@ import (
 	"github.com/vmware/go-vcloud-director/v2/util"
 )
 
-// SetApiToken behaves similarly to SetToken, with the difference that it will
-// return full information about the bearer token, so that the caller can make decisions about token expiration
-func (vcdClient *VCDClient) SetApiToken(org, apiToken string) (*types.ApiTokenRefresh, error) {
-	tokenRefresh, err := vcdClient.GetBearerTokenFromApiToken(org, apiToken)
+// TODO Have distinct names for API and Refresh tokens
+// Token is a struct that handles two methods: Delete() and GetInitialRefreshToken()
+type Token struct {
+	Token  *types.Token
+	client *Client
+}
+
+// CreateToken is used for creating API tokens and works in two steps:
+// 1. Register the token through the `register` endpoint
+// 2. Fetch it using GetTokenById(tokenID)
+// The user then can use *Token.GetInitialRefreshToken to get the API token
+func (vcdClient *VCDClient) CreateToken(org, tokenName string) (*Token, error) {
+	client := vcdClient.Client
+	if client.APIVCDMaxVersionIs("< 36.1") {
+		version, err := client.GetVcdFullVersion()
+		if err == nil {
+			return nil, fmt.Errorf("minimum version for API token is 10.3.1 - Version detected: %s", version.Version)
+		}
+		// If we can't get the VCD version, we return API version info
+		return nil, fmt.Errorf("minimum API version for API token is 36.1 - Version detected: %s", client.APIVersion)
+	}
```

Comment:
The first API call this function does (`GetTokenById`) will perform version validation

Reply:
Fixed.